From f6dddc2ac2e45f0b89d81a43d93ded50a1990b65 Mon Sep 17 00:00:00 2001
From: Alfred Chen <cchalpha@gmail.com>
Date: Wed, 8 Apr 2020 22:00:31 +0800
Subject: [PATCH 087/112] bmq: Fix sync-up error when BMQ applied but not
 enabled.

This should be apply to c36a1f520471 BMQ 0.97 in next kernel release.
---
 kernel/sched/sched.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index 299f3d94f64f..32e4177aeea3 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -2537,6 +2537,19 @@ static inline void membarrier_switch_mm(struct rq *rq,
 }
 #endif
 
+#ifdef CONFIG_SMP
+static inline bool is_per_cpu_kthread(struct task_struct *p)
+{
+	if (!(p->flags & PF_KTHREAD))
+		return false;
+
+	if (p->nr_cpus_allowed != 1)
+		return false;
+
+	return true;
+}
+#endif
+
 void swake_up_all_locked(struct swait_queue_head *q);
 void __prepare_to_swait(struct swait_queue_head *q, struct swait_queue *wait);
 
-- 
2.27.0.112.g101b3204f3

